### ISSUE TITLE
Progress for concat return wrong %

### ISSUE THREAD

**Issue by gabrielstuff** [fluent-ffmpeg/node-fluent-ffmpeg#361]
Using a simple : 

```
fluentffmpeg('./video_intro.mp4')
  .input('./opaskp.mp4')
  .input('./video_outro.mp4')
  .on('error', function(err) {
    console.log('An error occurred: ' + err.message);
  })
  .on('end', function() {
    console.log('Merging finished !');
  })
  .on('progress', function(progress) {
    console.log('Processing: ' + progress.percent + '% done');
  })
  .videoCodec('libx264')
  .mergeToFile('./full_movie.mp4', './temp');
```

`progress.percent` return `827.4509803921568%` at end 

Something is apparently wrong.

btw I love the lib :)

**Comment by njoyard** [comment]
Yeah, that's because percent is computed as #frames processed / #frames of first input. I don't think there's a way to properly compute percent in that case, but I'll have a look anyways.

**Comment by menocreativity** [comment]
Is there any news on this front?
From what I understand from your response is that we just need to calculate the number of frames in all queued videos and then use that number when calculating percentages.

It doesn't *sound* that complicated but my limited understanding of what is going on behind the scenes might be the reason for that ;)

**Comment by crosofg** [comment]
```
var keys = Object.keys(progress);
    console.log(keys)

[ 'frames', 'currentFps', 'currentKbps', 'targetSize', 'timemark' ]
```

**Comment by mirageisme** [comment]
`progress.percent` is undefined in my case. there is not errors in the console

**Comment by TomKaltz** [comment]
Here is what is returned for me.  I would really like to get the percentage completed when only a duration is returned from ffprobe

```
codecData { format: 'mov,mp4,m4a,3gp,3g2,mj2',
  audio: 'pcm_s24le (in24 / 0x34326E69)',
  video: 'dnxhd (AVdn / 0x6E645641)',
  duration: '00:00:44.21',
  video_details:
   [ 'dnxhd (AVdn / 0x6E645641)',
     'yuv422p(tv',
     'bt709/unknown/unknown)',
     '1920x1080',
     '145344 kb/s',
     '29.97 fps',
     '29.97 tbr',
     '2997 tbn',
     '2997 tbc (default)' ],
  audio_details:
   [ 'pcm_s24le (in24 / 0x34326E69)',
     '48000 Hz',
     '2 channels',
     's32 (24 bit)',
     '2304 kb/s (default)' ] }

progress { frames: 49,
  currentFps: 0,
  currentKbps: 0.2,
  targetSize: 0,
  timemark: '00:00:01.77' }
```

**Comment by mrnocreativity** [comment]
Oh, I totally forgot about this.

So what I did to calculate a decent percentage was to calculate percentages myself. As I have understood from the authors feedback, this percentage value only works correctly for processes where only 1 source video is involved. The process will however return the number of frames processed (even if multiple sources are involved).

So all we need is the total frame count to solve this.

Just loop over your sources before you start concatenating them and count the number of frames for every source file (using ffprobe). That's your total_frame_count. Now you can start the concatenation process. Upon every 'progress' event, you read the frames_processed value from the event and calculate the percentage based on the total_frame_count you got earlier.

It's a super hacky way but for the sake of continuing your project, this will probably help you out ;)

PS: It's been a while since I used this in my project, so I don't remember the exact details but you should probably be able to figure out where to go from here ;)

Good luck :)

**Comment by TomKaltz** [comment]
This is single source.  OSX ffmpeg 3.0.

**Comment by njoyard** [comment]
@TomKaltz can you open a separate issue then ? And paste your code ? Thanks :)

**Comment by TomKaltz** [comment]
Seems to be a specific file.  I will test further and report back.

**Comment by balajee** [comment]
any update on this?

**Comment by A113n1003** [comment]
any update?

**Comment by MartinBarker** [comment]
any update? I get very large, misleading percentages too
